dali/pipeline/operators/reader/loader/file_loader.cc

```diff
@@ -33,19 +32,20 @@ inline void assemble_file_list(const std::string& path, const std::string& curr_
 
   while ((entry = readdir(dir))) {
     std::string full_path = curr_dir_path + "/" + std::string{entry->d_name};
-    struct stat s;
-    stat(full_path.c_str(), &s);
-    if (S_ISREG(s.st_mode)) {
-      std::string rel_path = curr_entry + "/" + std::string{entry->d_name};
-      std::string file_name_lowercase = std::string{entry->d_name};
-      std::transform(file_name_lowercase.begin(), file_name_lowercase.end(),
-                     file_name_lowercase.begin(), ::tolower);
-      for (const std::string& s : valid_extensions) {
-        size_t pos = file_name_lowercase.rfind(s);
-        if (pos != std::string::npos && pos + s.size() == file_name_lowercase.size()) {
-          file_label_pairs->push_back(std::make_pair(rel_path, label));
-          break;
-        }
+#ifdef _DIRENT_HAVE_D_TYPE
+    if (entry->d_type != DT_REG && entry->d_type != DT_UNKNOWN) {
```

Comment:
Add `entry->d_type != DT_LNK` ?

Reply:
> Add `entry->d_type != DT_LNK` ?

Wouldn't this be outside of scope for this PR? To be considered in new one?

Reply:
Done